app/gui2/src/components/ComponentBrowser.vue

```diff
@@ -299,97 +238,33 @@ const selectedSuggestionIcon = computed(() => {
   return selectedSuggestion.value ? suggestionEntryToIcon(selectedSuggestion.value) : DEFAULT_ICON
 })
 
-watch(selectedPosition, (newPos) => {
-  if (newPos == null) return
-  highlightPosition.value = newPos
-})
-
-const highlightClipPath = computed(() => {
-  let height = animatedHighlightHeight.value
-  let position = animatedHighlightPosition.value
-  let top = position + ITEM_SIZE - height
-  let bottom = listContentHeight.value - position - ITEM_SIZE
-  return `inset(${top}px 0px ${bottom}px 0px round 16px)`
-})
-
-function selectWithoutScrolling(index: number) {
-  const scrollPos = scrolling.scrollPosition.value
-  scrolling.targetScroll.value = { type: 'offset', offset: scrollPos }
-  selected.value = index
-}
-
 // === Preview ===
 
-type PreviewState = { expression: string; suggestionId?: SuggestionId }
-const previewed = debouncedGetter<PreviewState>(() => {
-  if (selectedSuggestionId.value == null || selectedSuggestion.value == null) {
-    return { expression: input.code.value }
-  } else {
-    return {
-      expression: input.inputAfterApplyingSuggestion(selectedSuggestion.value).newCode,
-      suggestionId: selectedSuggestionId.value,
-    }
-  }
-}, 200)
+const previewed = debouncedGetter<string>(() => input.code.value, 200)
 
 const previewedSuggestionReturnType = computed(() => {
-  const id = previewed.value.suggestionId
+  const id = appliedSuggestion.value
```

Comment:
Because `appliedSuggestion` is only set in component-browsing mode, when editing an existing component we will lack information to select the correct visualization type